src/main/java/edu/cornell/kfs/concur/businessobjects/ValidationResult.java

```diff
@@ -43,15 +44,25 @@ public void addMessage(String message) {
         if (messages == null) {
             messages = new ArrayList<String>();
         }
-        messages.add(message);
+        if (StringUtils.isNotBlank(message) && isNotDuplicateMessage(message)) {
+            messages.add(message);
+        }
+    }
+    
+    private boolean isNotDuplicateMessage(String message) {
+        for (String currentMessage : getMessages()) {
```

Comment:
This function is presuming that the method caller is verifying "messages" is not null prior to being called which is ok for these changes as that is happening but if this private method is used elsewhere in this class in the future that may pose an issue.